components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java

```diff
@@ -103,7 +103,7 @@ private void setNameIDFormat(SPSSODescriptor spssoDescriptor, SAMLSSOServiceProv
     }
 
     private void setClaims(SPSSODescriptor spssoDescriptor, SAMLSSOServiceProviderDO samlssoServiceProviderDO) {
-        List<AttributeConsumingService> services = new ArrayList<>();
+        List<AttributeConsumingService> services;
         services = spssoDescriptor.getAttributeConsumingServices();
         if (services != null && services.size() > 0) {
             //assuming that only one AttrbuteComsumingIndex exists
```

Comment:
The following for loop is does nothing, so no use. Else clause is also of no use.
Basically, this whole method is doing nothing useful.
Either can remove this method if there are no usages, 
or should refine the method such that it does the intended work